lib/matplotlib/text.py

```diff
@@ -1158,6 +1159,10 @@ def set_rotation(self, s):
             The rotation angle in degrees in mathematically positive direction
             (counterclockwise). 'horizontal' equals 0, 'vertical' equals 90.
         """
+        if isinstance(s, str):
+            _api.check_in_list(['vertical', 'horizontal'], s=s)
```

Comment:
```suggestion
            _api.check_in_list(['vertical', 'horizontal'], rotation=rotation)
```
so that the message in the above test will  read

> 'invalid string' is not a valid value for rotation

Same with `_check_isinstance()` below.

While `s` is the exact parameter name, it's just a shortcut in this single-value setter. With "rotation" the error message is still understandable for direct calls `set_rotation()` (basically even more understandable because you likely only use that parameter positionally and don't even know that it's "s"). And it's also understandable if `rotation="foo"` is used in the constructor.